### PR TITLE
iOS,macOS: Clean up create_ios_framework.py

### PR DESCRIPTION
**Pull request by cbracken**
Over time, this script and others in sky/tools have accumulated a lot of additional and sometimes duplicate code.

This is a first pass cleanup of create_ios_framework.py to extract common utility code to utils.py and refactor for better readability.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cbracken** [comment]
This will be followed by cleanups for the other scripts. There's a lot more that could be cleaned up here, but I figured this patch already touches nearly half the lines of code in the script as-is.

File: sky/tools/create_ios_framework.py [hunk 1]
```diff
@@ -9,20 +9,14 @@
...
-import shutil
-import subprocess
 import sys
 
 from create_xcframework import create_xcframework  # pylint: disable=import-error
-
-ARCH_SUBPATH = 'mac-arm64' if platform.processor() == 'arm' else 'mac-x64'
-DSYMUTIL = os.path.join(
-    os.path.dirname(__file__), '..', '..', 'buildtools', ARCH_SUBPATH, 'clang', 'bin', 'dsymutil'
+from sky_utils import (  # pylint: disable=import-error
```
Comment (zanderso) [inline sky/tools/create_ios_framework.py:15]: Is there any way to do these imports in a way that requires retaining the library prefix (like `sky_utils.`) on usages?

**Review by zanderso — APPROVED** [review]

File: sky/tools/create_ios_framework.py (hunk 1 above)
Comment (cbracken) [reply sky/tools/create_ios_framework.py:15]: Yeah I could just import as `import sky_tools` and use the prefixed names everywhere. The list is kind of long at this point.

Comment (cbracken) [reply sky/tools/create_ios_framework.py]: Done.

**Pull request merged**